src/wp-includes/https-detection.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * HTTPS detection functions.
+ *
+ * @package WordPress
+ * @since 5.7.0
+ */
+
+/**
+ * Checks whether the website is using HTTPS.
+ *
+ * This is based on whether the home and site URL are using HTTPS.
+ *
+ * @since 5.7.0
+ *
+ * @return bool True if using HTTPS, false otherwise.
+ */
+function wp_is_using_https() {
+	if ( 'https' !== wp_parse_url( home_url(), PHP_URL_SCHEME ) ) {
+		return false;
+	}
+
+	// Use direct option access for 'siteurl' and manually run the 'site_url'
+	// filter because site_url() will adjust the scheme based on what the
+	// current request is using.
+	/** This filter is documented in wp-includes/link-template.php */
+	$site_url = apply_filters( 'site_url', get_option( 'siteurl' ), '', null, null );
+
+	if ( 'https' !== wp_parse_url( $site_url, PHP_URL_SCHEME ) ) {
+		return false;
+	}
+
+	return true;
+}
+
+/**
+ * Checks whether HTTPS is supported for the server and domain.
+ *
+ * @since 5.7.0
+ *
+ * @return bool True if HTTPS is supported, false otherwise.
+ */
+function wp_is_https_supported() {
+	$https_detection_errors = get_option( 'https_detection_errors' );
+
+	// If option has never been set by the Cron hook before, run it on-the-fly as fallback.
+	if ( false === $https_detection_errors ) {
+		wp_update_https_detection_errors();
+
+		$https_detection_errors = get_option( 'https_detection_errors' );
```

Comment:
Maybe it would be more ergonomic for this function to return the `$support_errors->errors`. This would avoid the need to re-get the option.

```suggestion
		$https_detection_errors = wp_update_https_detection_errors();
```

Reply:
In fact, maybe `wp_update_https_detection_errors()` should have this fallback detection inside of it if a `$use_cached` argument is passed? Then this method could just do:

```php
return empty( wp_update_https_detection_errors( true ) );
```

Reply:
I stuck with existing core patterns here of using this function only to update the cache. IMO since it's a cron hook (which therefore runs on an action) returning something might be confusing in terms of expectation.

I think the `get_option` call is safer to happen either way (whether inside or outside of `wp_update_https_detection_errors()`) because of all the hooks running during that process.